incubator/hydra/templates/hydra_deployment.yaml

```diff
@@ -0,0 +1,69 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
```

Comment:
1.  We've seen before that `beta` Kubernetes resources are fairly stable, I have no problem with making Hydra a StatefulSet.  I don't fully understand what the benefit of using StatefulSets would be in this case

If I understand correctly, the only thing you really need persistent storage for is the SSL certificates and CA files created by Hydra, right?

2. As for Postgres, it is currently a dependency, as @20zinnm said.  I followed in Gitlab's example, which may not have been the best idea considering Gitlab has the "omnibus" methodology of installing their software, but they also have a hell of a lot more than just Postgres as requirements.

There's a solution to this problem in Helm already.

By default, we will install the bundled postgres.  In `requirements.yaml`, I'll add a `condition: postgres.bundled`, and if that value is `true` in `values.yaml`, then the bundled postgres will be installed.

This way, users that will prefer using RDS can opt-in to using RDS, and those of us that don't care or prefer keeping everything in Kubernetes can keep it the way it is.

It makes sense to me to make external database connections opt-in because it's always going to require configuration anyways.  Keeping it bundled means the application will work out of the box without as much configuration.